### PR TITLE
Add support for setting default-repo in global config

### PR DESCRIPTION
**Pull request by nkubala**
This PR enables users to set the "default-repo" value in the skaffold
global config. This value will be substituted into all provided images
in the skaffold.yaml and k8s manifests according to certain
heuristics, allowing a better user experience for developing
applications against different image registries.

### CODE REVIEW

**Comment by balopat** [comment]
Another couple of thoughts: 

1. *precendence rules*: we were planning to introduce the flag based and environment variable based setting of `default-repo`. We might want to prepare in the design to represent the precedence rules of flag >  env-var > config-file/context-level > config-file/global 

2. *global config" name: I keep thinking what would be a better name, as this collides with the "skaffold config" that is describing the pipeline. Maybe this is a real config and the other one should be renamed to "pipeline"? other ideas instead of "global config":  "settings", "env",  "context", "preferences", "execution context", "runtime env". I like "settings", as it's similar to maven's "setting.xml" that there is a user level default location, but you can pass the location of it as well.

**Comment by nkubala** [comment]
@balopat thanks for the feedback!

> setting the default_repo on a global level doesn't work

Can you elaborate? It seems to work for me. `skaffold config set default-repo <repo_value> --global`

> no way to "unset" the default repo

I'll fix this in a separate PR.

> there is no "resolution" logic yet for when it is defined for a context as well globally

This part is true, I was weighing options on the best approach for this. I think the right answer here for order of precedence is:
1) Use the CLI flag: `--default-repo=<repo_value>` (fyi this will be implemented shortly)
2) Use the context-specific config value:
```
kube-context: gke_nkubala-demo_us-west1-a_demo-cluster
default-repo: gcr.io/nkubala-demo
```
3) Fallback to the global default-repo value, if set
```
global:
  default-repo: gcr.io/k8s-skaffold
```
4) Do nothing

WDYT?

RE: naming, I'm also of the mind that we should considering renaming the `SkaffoldConfig` instead of renaming this. `SkaffoldSpec` or `SkaffoldPipeline` would be my two votes. If we renamed this, I'd considering calling it either `settings` or `preferences`.

File: cmd/skaffold/app/cmd/config/util.go [hunk 1]
```diff
@@ -59,7 +60,10 @@ func resolveConfigFile() error {
 	return util.VerifyOrCreateFile(configFile)
 }
 
+// ReadConfigForFile reads the specified file and returns the contents
+// parsed into a Config struct.
 func ReadConfigForFile(filename string) (*Config, error) {
+	fmt.Printf("reading config for file %s\n", filename)
```
Comment (balopat) [inline cmd/skaffold/app/cmd/config/util.go]: ux-nit: this feels like a bit of clutter on the screen - can we make it available only for debug logging?

**Comment by balopat** [comment]
I'll open a separate PR to address my nits, as I can't add to your branch.

**Pull request merged**